spec/serializers/notification_serializer_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe NotificationSerializer do
+  subject(:serializer) { described_class.new(notification) }
+
+  let(:notification) { create(:notification) }
+
+  describe 'move?' do
+    it { expect(serializer.move?).to be true }
+  end
+
+  describe 'move_url' do
+    it { expect(serializer.move_url).to eql "http://localhost:4000/api/v1/moves/#{notification.topic.id}" }
+  end
+
+
+  describe 'json rendering' do
+    let(:adapter_options) { {} }
+    let(:result) do
+      JSON.parse(ActiveModelSerializers::Adapter.create(serializer, adapter_options).to_json).deep_symbolize_keys
+    end
+
+    let(:expected_result) {
+      {
+          "data": {
+              "id": notification.id,
+              "type": 'notifications',
+              "attributes": {
+                  "event_type": 'move_created',
+                  "timestamp": notification.created_at.as_json,
+              },
+              "relationships": {
+                  "move": {
+                      "data": {
+                          "id": notification.topic.id,
+                          "type": 'moves',
+                      },
+                      "links": {
+                          "self": "http://localhost:4000/api/v1/moves/#{notification.topic.id}",
+                      },
+                  },
+              },
+          },
+      }
+    }
+
+    it 'contains a type property' do
+      expect(result[:data][:type]).to eql 'notifications'
+    end
+
+    it 'contains an id property' do
+      expect(result[:data][:id]).to eql notification.id
+    end
+
+    it 'contains a event_type attribute' do
+      expect(result[:data][:attributes][:event_type]).to eql 'move_created'
+    end
+
+    it 'contains a timestamp attribute' do
+      expect(result[:data][:attributes][:timestamp]).to eql notification.created_at.as_json
+    end
+
+    it 'contains a move relationship data' do
+      expect(result[:data][:relationships][:move][:data]).to eql(id: notification.topic.id, type: 'moves')
+    end
+
+    it 'contains a move relationship links' do
+      expect(result[:data][:relationships][:move][:links]).to eql(self: "http://localhost:4000/api/v1/moves/#{notification.topic.id}")
+    end
+
+    it 'renders the correct json' do
+      expect(result).to eql(expected_result)
```

Comment:
should this be the only test necessary? 🤔

Reply:
Can't speak for @martyn-w but I believe it's checking that the actual result is what he's expecting (and he built "manually" inside the `let`)

Reply:
hi @ngw @alexdesi ... true, there is a little duplication here. I had intended the first set of tests to check the values, and this test to check the shape of the json (in particular, getting the <links> object inside of the <relationships> object. But whether all tests are strictly necessary I'll leave to your better judgement :-)

Reply:
👍  for me, I'm not strongly opinionated on this.